internal/impl/gcp/processor_vertex_ai.go

```diff
@@ -0,0 +1,316 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed as a Redpanda Enterprise file under the Redpanda Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// https://github.com/redpanda-data/connect/blob/main/licenses/rcl.md
+
+package gcp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"unicode/utf8"
+
+	"cloud.google.com/go/vertexai/genai"
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"google.golang.org/api/option"
+)
+
+const (
+	vaiFieldProject          = "project"
+	vaiFieldCredentialsJSON  = "credentials_json"
+	vaiFieldModel            = "model"
+	vaiFieldLocation         = "location"
+	vaiFieldPrompt           = "prompt"
+	vaiFieldSystemPrompt     = "system_prompt"
+	vaiFieldTemp             = "temperature"
+	vaiFieldTopP             = "top_p"
+	vaiFieldTopK             = "top_k"
+	vaiFieldMaxTokens        = "max_tokens"
+	vaiFieldStop             = "stop"
+	vaiFieldPresencePenalty  = "presence_penalty"
+	vaiFieldFrequencyPenalty = "frequency_penalty"
+	vaiFieldResponseFormat   = "response_format"
+)
+
+func init() {
+	err := service.RegisterProcessor(
+		"gcp_vertex_ai_chat",
+		newVertexAIProcessorConfig(),
+		newVertexAIProcessor,
+	)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func newVertexAIProcessorConfig() *service.ConfigSpec {
+	return service.NewConfigSpec().
+		Categories("AI").
+		Summary("Generates responses to messages in a chat conversation, using the Vertex AI API.").
+		Description(`This processor sends prompts to your chosen large language model (LLM) and generates text from the responses, using the Vertex AI API.
+
+For more information, see the https://cloud.google.com/vertex-ai/docs[Vertex AI documentation^].`).
+		Version("4.33.0").
```

Comment:
Guess we'll have to bump this to `4.34.0` or not sure what will be the next release.

Reply:
I've bumped it! I wish there was a way to automate this somehow... 🤔

Reply:
Yeah, indeed. It would be a good investment to build some automation. Maybe we can use `redpanda-connect list --format jsonschema` to figure out which components / fields are new. Should think about that and chat with Ash when he's around.